lib/pbench/server/api/resources/datasets_metadata.py

```diff
@@ -181,9 +181,9 @@ def _put(self, params: ApiParams, _) -> Response:
                 Metadata.setvalue(key=k, value=v, dataset=dataset, user_id=user_id)
             except MetadataError as e:
                 self.logger.warning("Unable to update key {} = {!r}: {}", k, v, str(e))
-                fail = True
+                fail.append(str(e))
```

Comment:
Looking at `lib/pbench/test/unit/server/test_datasets_metadata.py` line 435, it seems like the value of `str(e)` already contains the key and value...if that's good enough at line 184, do we really need all that extra stuff at line 183?

Reply:
I didn't really look at 183. If I were going to revise it, I might just drop the entire line and possibly put out a single `warning` afterwards using the list. I don't think that's worth another revision: but I suppose I asked for this by taking the bait for some opportunistic refactoring. 😆